build.cake

```diff
@@ -0,0 +1,81 @@
+// Arguments.
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+
+// Define directories.
+var solutionFile = GetFiles("./Okta.Xamarin/*.sln").First();
+
+// Android and iOS.
+var androidProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.Android/*.csproj").First();
+var iOSProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.iOS/*.csproj").First();
+
+// Output folders.
+var artifactsDirectory = Directory(System.IO.Path.Combine(Environment.CurrentDirectory, "artifacts"));
+var androidOutputDirectory = Directory(System.IO.Path.Combine(artifactsDirectory, "Android"));
+var iOSOutputDirectory = Directory(System.IO.Path.Combine(artifactsDirectory, "iOS"));
+
+// Tests.
+var testsProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.Test/*.csproj").First();
+
+Task("Clean")
+	.Does(() => 
+	{
+		CleanDirectory(artifactsDirectory);
+
+		DotNetBuild(solutionFile, settings => settings
+			.SetConfiguration(configuration)
+			.WithTarget("Clean")
+			.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Restore-Packages")
```

Comment:
Shouldn't this task depend on `clean`?

Reply:
I wanted to treat this as a kind of "atomic" operation.  The build tasks output to the same place, if "Restore-Packages" cleaned then subsequent operations could delete previous output.